Library/Homebrew/livecheck/skip_conditions.rb

```diff
@@ -0,0 +1,228 @@
+# typed: true
+# frozen_string_literal: true
+
+require "livecheck/livecheck"
+
+module Homebrew
+  module Livecheck
+    # The `Livecheck::SkipConditions` module primarily contains methods that
+    # check for various formula/cask conditions where a check should be skipped.
+    #
+    # @api private
+    module SkipConditions
+      extend T::Sig
+
+      module_function
+
+      sig do
+        params(
+          formula_or_cask: T.any(Formula, Cask::Cask),
+          livecheckable:   T::Boolean,
+          full_name:       T::Boolean,
+          verbose:         T::Boolean,
+        ).returns(T.nilable(Hash))
+      end
+      def formula_or_cask_skip(formula_or_cask, livecheckable, full_name: false, verbose: false)
+        formula = formula_or_cask if formula_or_cask.is_a?(Formula)
+
+        stable_from_google_code_archive = formula&.stable&.url&.match?(
+          %r{https?://storage\.googleapis\.com/google-code-archive-downloads/}i,
+        )
+        stable_from_internet_archive = formula&.stable&.url&.match?(%r{https?://web\.archive\.org/}i)
+        stable_is_gist = formula&.stable&.url&.match?(%r{https?://gist\.github(?:usercontent)?\.com/}i)
```

Comment:
Would be good to avoid the need for all the `&` (particularly those that are repeated) by checking the values and making more logic conditional.